Improve admin_default navbar & responsiveness

BelleNottelling opened the pull request:
Reworks our admin panel navbar to follow the original design of Tabler, and is now much more responsive. I believe this improves usability for devices of all sizes. Also closes #710
1440p:
![image](https://user-images.githubusercontent.com/17304943/225027321-210b84cf-6860-4188-838a-a7a47975945c.png)
1080p:
![image](https://user-images.githubusercontent.com/17304943/225027476-95005c45-488d-47e4-8e3d-ac61afcc89fa.png)
720p:
![image](https://user-images.githubusercontent.com/17304943/225027547-1390aa8c-9e0c-44ae-9544-9516baff5217.png)
Mobile:
![image](https://user-images.githubusercontent.com/17304943/225027653-bf221bd4-3494-41cd-9836-8305634a83bb.png)

admdly reviewed (approved):
Excellent improvement in UI, imho. 👍

evrifaessa commented on src/themes/admin_default/html/layout_default.html.twig:
```diff
@@ -116,91 +133,67 @@
+                             {% if group.subpages %}
+                                <li class="nav-item dropdown">
+                                    <a class="nav-link dropdown-toggle" href="#navbar-base" data-bs-toggle="dropdown" data-bs-auto-close="outside" role="button" aria-expanded="false" >
+                                        <svg class="icon icon-tabler icon-tabler-arrow-back-up only-large-screens">
```
The icon and the text needs a little space in between.

jaapmarcus reviewed (commented):
With some screen sizes the top menu doesn't go in a hamburger fast enough:
<img width="859" alt="Screenshot 2023-03-14 at 11 37 34" src="https://user-images.githubusercontent.com/9754650/224975620-98c25efe-0f43-427c-856d-b46d8050ab3f.png">

It also mean we limit the number menu item we can handle in the top..

BelleNottelling commented:
Thanks, @jaapmarcus & @evrifaessa
I've addressed your reviews, things a definitely a bit better now.

Jaap, I've changed the breakpoint for the hamburger menu to the XL size, so it'll collapse down at 1200px, which takes care of the issue for me with all possible modules enabled. And yes, this will unfortunately limit the number of menu items we can have, but I still think this is a big improvement over what's already existed, and there's some existing modules that can be moved to be a bit more sensible, such as moving the KB under the support category which will free up more space.

![image](https://user-images.githubusercontent.com/17304943/225022489-3f3d95df-d3a3-41d6-b5d5-8763d3547fdd.png)
![image](https://user-images.githubusercontent.com/17304943/225022565-326e86fa-b5ef-4592-928a-8265a2c1842a.png)

John-S4 commented:
> And yes, this will unfortunately limit the number of menu items we can have

The logical solution for that is for further menu items to be grouped under a 'More...' item, but I don't think that is in the scope of this PR really. 

The padding on the source and sponsor buttons looks very odd on the 720px screenshot, there is much more gap inside the box on the right than the left. 

Personally I would much rather not have the country code displayed in the language dropdown, because I think the flag and language name are sufficient and I don't see it adds anything. 

Apart from those little thing though, it's a massive improvement and I prefer it to any of the other proposed options with a sidebar stuck out to the left and huge weird gaps between content on bigger screens.

John-S4 reviewed (approved):
Approved, but also left some comments in, well, a comment. :)}

admdly commented:
> Personally I would much rather not have the country code displayed in the language dropdown, because I think the flag and language name are sufficient and I don't see it adds anything.

I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?

That said, there's some improvements to be made with i18n in general, at which point it could be addressed rather than in this PR.

BelleNottelling commented:
> I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?

The flag icons, but I don't know if they will always differ, but even then, the displayed name should be different
![image](https://user-images.githubusercontent.com/17304943/225057901-75af69f3-8315-4ee2-ace3-652f3dfb922d.png)

John-S4 commented:
> I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?

Well, the language name will be different. The flag will indicate the country, and the language name the actual language. 

There are a lot of websites out there with language switchers and I'm not sure I've ever seen another one with the code included as well as flag/name.

admdly commented:
> > I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?
> 
> 
> 
> Well, the language name will be different. The flag will indicate the country, and the language name the actual language. 
> 
> 
> 
> There are a lot of websites out there with language switchers and I'm not sure I've ever seen many others with the code included as well as flag/name.

With English, you do often have English (US), English (UK), etc. but again, I agree this is a minor issue at this point that can be resolved with better i18n. 

The full language code is unnecessary for most languages and the country code should only be shown when clarification is needed.

BelleNottelling commented:
Okay, I fixed the icon spacing John noticed. I think any changes to the locale selector can be considered out of the scope for the PR, especially since it seems like we still need to have a bit of a discussion and come to a universal agreement on it
![image](https://user-images.githubusercontent.com/17304943/225066610-0acd5931-b994-49ba-a58c-6329d8134f64.png)

John-S4 commented:
Fine to merge as far as I'm concerned.

shmshd commented:
![image](https://user-images.githubusercontent.com/11484121/225204399-c921cdf3-4f52-4a9f-937f-ac4e7125d861.png)

Logo is not switched between light/dark.

shmshd commented on src/themes/admin_default/html/layout_default.html.twig:
```diff
@@ -62,17 +89,17 @@
                                  <use xlink:href="#brand-github" />
                               </svg>
-                              {{ 'Source code'|trans }}
+                              <span class="only-large-screens">{{ 'Source code'|trans }}</span>
```
I think it is best to show the text from 1024px onwards as there is quite enough space for both icon texts.

shmshd commented:
![image](https://user-images.githubusercontent.com/11484121/225206686-fbbf41ec-ed36-4a28-85ee-ac41dbfe52e7.png)

Equal gaps?
